pg_wait_sampling.c

```diff
@@ -92,6 +94,10 @@ get_max_procs_count(void)
 {
 	int count = 0;
 
+#if PG_VERSION_NUM >= 150000
+	Assert(MaxBackends > 0);
+	count += MaxBackends;
+#else
 	/*
 	 * MaxBackends: bgworkers, autovacuum workers and launcher.
 	 * This has to be in sync with the value computed in
```

Comment:
The content of caution comment for `get_max_procs_count()` have to be moved to this one.

Reply:
Ah indeed!

I reworked the comments to be extra clear about why there are so many different parts in the function and what part is responsible for which part of the final value.